src/frameworks/uikit/ui_device.rs

```diff
@@ -14,9 +16,49 @@ pub const UIDeviceOrientationPortrait: UIDeviceOrientation = 1;
 #[allow(dead_code)]
 pub const UIDeviceOrientationPortraitUpsideDown: UIDeviceOrientation = 2;
 pub const UIDeviceOrientationLandscapeLeft: UIDeviceOrientation = 3;
-#[allow(dead_code)]
 pub const UIDeviceOrientationLandscapeRight: UIDeviceOrientation = 4;
 #[allow(dead_code)]
 pub const UIDeviceOrientationFaceUp: UIDeviceOrientation = 5;
 #[allow(dead_code)]
 pub const UIDeviceOrientationFaceDown: UIDeviceOrientation = 6;
+
+#[derive(Default)]
+pub struct State {
+    current_device: Option<id>,
+    system_version: Option<id>,
+}
+
+pub const CLASSES: ClassExports = objc_classes! {
+
+  (env, this, _cmd);
+
+  @implementation UIDevice: NSObject
+
+  + (id)currentDevice {
+    if let Some(device) = env.framework_state.uikit.ui_device.current_device {
+      device
+    } else {
+      let new = env.objc.alloc_static_object(
+          this,
+          Box::new(TrivialHostObject),
+          &mut env.mem
+      );
+      env.framework_state.uikit.ui_device.current_device = Some(new);
+      new
+    }
+  }
+
+  - (id) systemVersion { // NSString
```

Comment:
Style nitpicks:

* I tend to not have a space between the `)` and the method name, but of course it's not important.

* Hmm, I think I'm only using comments to mark argument types rather than return types right now, but maybe I should change that. Perhaps return types could be on the preceding line so that they'd be clear in a case like:

```
// NSString
- (id)methodReturningString:(id)str { // NSString```